modules/calib3d/test/test_usac.cpp

```diff
@@ -0,0 +1,326 @@
+        const double conf = 0.99, thr = 2., max_iters = 1.2 * log(1 - conf) /
+                 log(1 - pow(inl_ratio, 4 /* sample size */));
+        cv::Mat mask, H = cv::findHomography(pts1, pts2,USAC_DEFAULT, thr, mask,
+                                                   int(max_iters), conf);
+        checkInliersMask(TestSolver::Homogr, inl_size, thr, pts1, pts2, H, mask);
+    }
+}
+
+TEST(usac_Homography_parallel, accuracy) {
+    std::vector<int> gt_inliers;
+    const int pts_size = 1500;
+    cv::RNG &rng = cv::theRNG();
+    for (double inl_ratio = 0.05; inl_ratio < 0.91; inl_ratio += 0.1) {
+        cv::Mat pts1, pts2, K1, K2;
+        int inl_size = generatePoints(rng,pts1, pts2, K1, K2, false /*two calib*/,
+          pts_size, TestSolver ::Homogr, inl_ratio, 0.2 /*noise std*/, gt_inliers);
+        const double conf = 0.99, thr = 2., max_iters = 1.5 * log(1 - conf) /
+                  log(1 - pow(inl_ratio, 4 /* sample size */));
+        cv::Mat mask, H = cv::findHomography(pts1, pts2,USAC_PARALLEL, thr, mask,
+                                                   int(max_iters), conf);
+        checkInliersMask(TestSolver::Homogr, inl_size, thr, pts1, pts2, H, mask);
+    }
+}
+
+TEST(usac_Fundamental, accuracy) {
+    std::vector<int> gt_inliers;
+    const int pts_size = 1500;
+    cv::RNG &rng = cv::theRNG();
+    // start from 20% otherwise max_iters will be too big
+    for (double inl_ratio = 0.2; inl_ratio < 0.91; inl_ratio += 0.1) {
+        cv::Mat pts1, pts2, K1, K2;
+        int inl_size = generatePoints(rng,pts1, pts2, K1, K2, false /*two calib*/,
+          pts_size, TestSolver ::Fundam, inl_ratio, 0.1 /*noise std*/, gt_inliers);
+        const double conf = 0.99, thr = 1., max_iters = 1.2 * log(1 - conf) /
+                    log(1 - pow(inl_ratio, 7 /* sample size */));
+        cv::Mat mask, F = cv::findFundamentalMat(pts1, pts2,USAC_DEFAULT, thr, conf,
+                             int(max_iters), mask);
+        checkInliersMask(TestSolver::Fundam, inl_size, thr, pts1, pts2, F, mask);
+    }
+}
+
+TEST(usac_Fundamental8pts, accuracy) {
+    std::vector<int> gt_inliers;
+    const int pts_size = 1500;
+    cv::RNG &rng = cv::theRNG();
+    for (double inl_ratio = 0.20; inl_ratio < 0.91; inl_ratio += 0.1) {
+        cv::Mat pts1, pts2, K1, K2;
+        int inl_size = generatePoints(rng,pts1, pts2, K1, K2, false /*two calib*/,
+                      pts_size, TestSolver ::Fundam, inl_ratio, 0.1 /*noise std*/, gt_inliers);
+        const double conf = 0.99, thr = 1., max_iters = 1.2 * log(1 - conf) /
+            log(1 - pow(inl_ratio, 8 /* sample size */));
+        cv::Mat mask, F = cv::findFundamentalMat(pts1, pts2,USAC_FM_8PTS, thr, conf,
+                                                       int(max_iters), mask);
+        checkInliersMask(TestSolver::Fundam, inl_size, thr, pts1, pts2, F, mask);
+    }
+}
+
+TEST(usac_Essential, accuracy) {
+#if defined(HAVE_EIGEN) || defined(HAVE_LAPACK)
```

Comment:
Why disable test body only? (exclude whole test)

What is the problem with Eigen/ Lapack? (add a note thorough comment)

Reply:
On Windows there is no Eigen and Lapack, however my essential solver assumes one of them - otherwise, it returns 0 solution every time. So, if I do not disable the test - it will fail.

Reply:
Please use Error with `StsNotImplemented` instead of 0 solution.
(you may catch that in tests)